README.md

```diff
@@ -1609,6 +1609,35 @@ runtime {
 
 This attribute is mandatory when submitting tasks to JES. When running on other backends, they default to not running the process within Docker.
 
+### Docker Tags
+
+Docker tags are a convenient way to point to a version of an image (ubuntu:14.04), or even the latest version (ubuntu:latest).
+For that purpose, tags are mutable, in a sense that the image they point to can change, while the tag name stays the same.
+While this is very convenient in some cases, using mutable, or "floating" tags in WDL affects the reproducibility of the WDL file: the same WDL using "ubuntu:latest" run now, and a year, or even a month from now will actually run with different docker images.
+This has an even bigger impact when Call Caching is turned on in Cromwell, and could lead to false positive cache hits.
+Docker provides another way of identifying an image version, using the specific digest of the image. The digest is guaranteed to be different if 2 images have the same byte content. For more information see https://docs.docker.com/registry/spec/api/#/content-digests
+A docker image with digest can be referenced as follow :
+    ubuntu@sha256:71cd81252a3563a03ad8daee81047b62ab5d892ebbfbf71cf53415f29c130950
+The above image refers to a specific image of ubuntu, that does not depend on a floating tag.
+A wdl containing this docker image run now and a year from now will run on the exact same image.
+
+In order to remove the possibility of false positive, Cromwell takes the following approach regarding floating docker tags.
+
+When Cromwell finds a job ready to be run, it will first look at its docker runtime attribute, and apply the following logic:
+- The job doesn't specify a docker image:
+    The job will be dispatched and all call caching settings (read/write) will apply normally.
+- The job does specify a docker runtime attribute:
+    - The docker image uses a hash (e.g. ubuntu@sha256:71cd81252a3563a03ad8daee81047b62ab5d892ebbfbf71cf53415f29c130950):
+        All call caching settings apply normally.
+    - The docker image uses a floating tag:
+        Cromwell will look up the digest of the image, and replace the user's docker value with the digest. This way cromwell can report with certainty which specific image was used to run a job.
+        However, because this is done dynamically when a job is about to run, odd scenarios can occur where a tag can be updated in the middle of a workflow.
```

Comment:
Let's say you have a really wide scatter and the tag gets updated in the middle of it. Half of the shards would get a hash and the other half another hash. Which means potentially half of them could have a cache hit but not the other..